### ISSUE TITLE
A11y: "Form control element <textarea> has no associated label"

### ISSUE THREAD

**Issue by TheeMattOliver** [FormidableLabs/react-live#334]
Hi all, I love `react-live` and it's been great building docs with it.

## Problem

When assessing WCAG 2.1 AA conformance in a project shipped with `react-live` using [this tool](https://chrome.google.com/webstore/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp?hl=en-US), the `textarea` used for live editing is failing [Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) here:

```html
<textarea
    style="margin: 0px; border: 0px; background: none; box-sizing: inherit; display: inherit; font-family: inherit; font-size: inherit; font-style: inherit; font-variant-ligatures: inherit; font-weight: inherit; letter-spacing: inherit; line-height: inherit; tab-size: inherit; text-indent: inherit; text-rendering: inherit; text-transform: inherit; white-space: pre-wrap; word-break: keep-all; overflow-wrap: break-word; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; resize: none; color: inherit; overflow: hidden; -webkit-font-smoothing: antialiased; -webkit-text-fill-color: transparent; padding: 16px;"
    data-gramm="false"
    spellcheck="false"
    autocorrect="off"
    autocomplete="off"
    autocapitalize="off"
    class="npm__react-simple-code-editor__textarea">
```

## Todo's

There are a few things the tools suggests:

- If the control is a [labelable](https://html.spec.whatwg.org/multipage/forms.html#category-label) element, add a for attribute to a visible label with the id of this control (e.g. <label for="controlID">);
- Or, if the control is a labelable element, put the control inside a visible <label> element;
- Or, add an aria-labelledby attribute to the control. It must point to visible text on the page that is meaningful as a label;
- Or, if the design cannot have a visible label, provide a label using the aria-label attribute (e.g. aria-label="Activities");
- Or, if the design cannot have a visible label, use the title attribute to provide a label.

## Proposed solution

Use the title attribute to provide a label according to [WAI technique 65](https://www.w3.org/WAI/WCAG22/Techniques/html/H65), whose objective is "to use the title attribute to provide an accessible name for form controls when the visual design does not include text on the screen that can be associated with the control as a label. User agents, including assistive technology, can speak the title attribute". 

## Who does this affect?
- People using a screen reader, including blind, low vision and neurodivergent people
- People with dexterity impairment using voice control

## Next steps

~~I am happy to open a PR, but I'm just wondering where this `textarea` is rendered since I don't see it in `react-live` or in `use-editable`~~

**Update**: this is rendered by [`react-simple-code-editor`](https://github.com/react-simple-code-editor/react-simple-code-editor), opening this up over there.
